src/vuln_analysis/utils/source_rpm_downloader.py

```diff
@@ -0,0 +1,731 @@
+
+import os
+from pathlib import Path
+import gzip
+import tarfile
+import xml.etree.ElementTree as ET
+import subprocess
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+from dataclasses import dataclass
+from typing import List
+import threading
+from vuln_analysis.data_models.input import SBOMPackage
+from vuln_analysis.logging.loggers_factory import LoggingFactory
+logger = LoggingFactory.get_agent_logger(__name__)
+
+
+@dataclass
+class RepoUrl:
+    url: str
+    platform: str  # e.g., 'el9', 'el10'
+
+
+def extract_archives_in_folder(folder: str):
+    """
+    Extracts all .tar.* files in the given folder to subdirectories.
+    Supports .tar.gz, .tgz, .tar.xz, .txz, .tar.bz2, .tbz2, and .tar.
+
+    :param folder: Path to the folder to scan.
+    """
+    supported_extensions = {
+        '.tar.gz', '.tgz', 
+        '.tar.xz', '.txz', 
+        '.tar.bz2', '.tbz2', 
+        '.tar'
+    }
+
+    folder_path = Path(folder)
+
+    for file in folder_path.iterdir():
+        if not file.is_file():
+            continue
+
+        suffix = ''.join(file.suffixes[-2:]) if len(file.suffixes) >= 2 else file.suffix
+        if suffix not in supported_extensions and file.suffix not in supported_extensions:
+            continue
+
+        # Determine full suffix like .tar.gz or .tgz
+        full_suffix = suffix if suffix in supported_extensions else file.suffix
+
+        # Determine output directory name
+        output_dir = file.with_suffix('').with_suffix('').stem if full_suffix.startswith('.tar') else file.stem
+        output_path = folder_path / output_dir
+
+        print(f"Extracting: {file.name} → {output_path}")
```

Comment:
@RedTanny Please replace print with logger.debug
```suggestion
        logger.debug(f"Extracting: {file.name} → {output_path}")
```